org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDocumentLifeCycleHandler.java

```diff
@@ -107,7 +119,7 @@ public BaseDocumentLifeCycleHandler(boolean delayValidation) {
 				public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
 					long start = System.currentTimeMillis();
 					IStatus status = performValidation(monitor);
-					movingAverage.update(System.currentTimeMillis() - start);
+					movingAverageForValidation.update(System.currentTimeMillis() - start);
```

Comment:
Just fyi as I have also seen it in other places. `System.currentTimeMillis` can go backward because it observes clock adjustements so this could get negative. Not sure if it's a problem here, but using `System.nanoTime()` is usually the better option to measure durations as it uses a monotonic clock instead of wall-clock.

Reply:
Thank you for the suggestion. 👍

`nanoTime()` is used now.